programs/survival/lava_sponge.sc

```diff
@@ -0,0 +1,43 @@
+//just a sponge work on lava like water
+//by _GieR
+
+
+__config() -> (
+   m(
+      l('stay_loaded','true')
+   )
+);
+
+
+__on_player_places_block(player, item_tuple, hand, block)->(
+    if(hand != 'mainhand' || item_tuple:0 != 'sponge' || all(neighbours(pos(block)),_!='lava'), return());
+    set(pos(block),'wet_sponge');
+    count = 0;
+    recursion = 0;
+    sponge(l(pos(block)), recursion, count);
+);
+
+sponge(coord_list, recursion, count)->(
+    lava = {};
+    for(coord_list,
+        coord = _;
+        for(neighbours(coord), 
+            if(_ == 'lava' && recursion<=7 && count<=64,
+                if(!has(lava, pos(_)),
+                    lava += pos(_);
+                    count +=1;
+                )
+            )      
+        )
+    );     
+    if(lava!={} && recursion<=7 && count<=65,

```

Comment:
i tried to run some script like
 `/script run a={}; c_for(d=0,d<1000,d+=1,a+=d); if(a!={}, print('false'))`
and
 `/script run a={}; c_for(d=0,d<1000,d+=1,a+=d); if(length(a)!=0, print('false'))`
their runtime are basically the same
not sure if this script can show the difference though

Reply:
well for me the first one took 38ms, and the second one took 0.4ms...
To compare them both, it would be best to have a function which adds say 1000 items to `a`, and then at each step do `profile_expr(a=={})` and `profile_expr(length(a)==0)`, and then see how they increase over time. Though off the top of my head I can tell you that a=={} will be O(n) cos you have to check each item, whereas length(a)==0 is designed to be O(1).

Reply:
I see, Ill use length instead